### PR TITLE
Build JniBridge with jdk11 and download SDK from stevedore

### PR DESCRIPTION
**Pull request by juliusmik**
* Update JDK used for building JNIBridge to JDK 11
* Download SDK from stevedore using bee instead of directly from Google servers using perl
* Download Google Play Services as a separate dependency, since it is only needed by the JNIBride and not used in other places where Android SDK is used

The tests fail to compile locally on my M1 mac. Building **Lib_Mac_x64 artifacts/testlib/debug_Mac_x64_nonlump/testlib.a** with **artifacts/Stevedore/mac-toolchain-10_13_e684/usr/bin/libtool** exits with error code 137, but no error messages. This doesn't seem to be related to JDK11 upgrade and the tests pass in yamato, so I'm not going to do anything in this PR, but we should keep this in mind, that we'll likely have to update bee version which uses osx tools compatible with M1.

### CODE REVIEW

File: JniBridge.bee.cs [hunk 1]
```diff
@@ -184,24 +185,31 @@ static Jdk SetupJava()
...
 
-    static NPath GetEnv(string variable)
+    static NPath SetupGooglePlayServices()
     {
-        var path = Environment.GetEnvironmentVariable(variable);
-        if (string.IsNullOrEmpty(path))
-            throw new Exception($"Environment variable {variable} not set");
-        return new NPath(path);
+        // Downloaded from https://dl.google.com/android/repository/google_play_services_3265130_r12.zip
+        var gps = StevedoreArtifact.Testing("google-play-services/3265130_r12_3a601d0d9366fe05c7a56855f2c906a8d703fd297f1c0f6cfdaf27b181be284c.zip");
```
Comment (juliusmik) [inline JniBridge.bee.cs]: I'll ask for the artifact to be coppied from the testing to the internal repository and I'll change Testing to UnityInternal before merging if everything here looks good.

File: JniBridge.bee.cs [hunk 2]
```diff
@@ -466,7 +474,7 @@ static NativeProgram SetupTestProgramWindows(ToolChain toolchain, NativeProgram
 
         var destDir = new NPath("build").Combine(Platform.Windows);
         var config = new NativeProgramConfiguration(codegen, toolchain, false);
-        var target = np.SetupSpecificConfiguration(config, config.ToolChain.ExecutableFormat).DeployTo(destDir);
+        var target = np.SetupSpecificConfiguration(config, config.ToolChain.ExecutableFormat).DeployTo(destDir);
```
Comment (juliusmik) [inline JniBridge.bee.cs:486]: I have no idea what  has changed in this line and the line below, but automatic code formatter changes something if this file is saved even without any actual changes.

File: apigenerator/APIGenerator.java [hunk 3]
```diff
@@ -20,31 +20,39 @@
...
-			System.err.format("Usage: APIGenerator <dst> <jarfile[;jarfile;...]> <regex...>\n");
+			System.err.format(k_UsageMessage);
 			System.err.format("%s: is not a directory.\n", dst);
 			System.exit(1);
 		}
-
-		String jarList = args.pollFirst();
+		String nextArgument = args.pollFirst();
+		boolean useSystemClasses = false;
+		if (nextArgument.equals("-s"))
```
Comment (aurimasc) [inline apigenerator/APIGenerator.java]: Better to use opposite for the null-safety:
if ("-s".equals(nextArgument))

File: apigenerator/APIGenerator.java [hunk 4]
```diff
@@ -122,11 +140,40 @@ public void collectDependencies(LinkedList<JarFile> files, LinkedList<String> ar
...
+					System.err.format("\nCan't find %s class!\n\n", arg);
+					continue;
+				}
+				try
+				{
+					Class c = ClassLoader.getSystemClassLoader().loadClass(convertToBinaryClassName(arg));
+					String cppClassName = getClassName(c);
+					int nClasses = m_DependencyChain.size();
+					collectDependencies(c);
+					System.err.format("[%d][%d]\t%s\n", m_DependencyChain.size(), m_DependencyChain.size() - nClasses, cppClassName);
```
Comment (aurimasc) [inline apigenerator/APIGenerator.java]: I see this is consistent with existing code, but why are we outputting to stderr here, when it looks more like an informational log message?

**Review by aurimasc — APPROVED** [review]

File: JniBridge.bee.cs (hunk 1 above)
Comment (juliusmik) [reply JniBridge.bee.cs]: Changed in https://github.com/Unity-Technologies/android-jni-bridge/pull/41/commits/cf5e1294fa955d4487bbebb8b2e4e946be544bc0

File: apigenerator/APIGenerator.java (hunk 3 above)
Comment (juliusmik) [reply apigenerator/APIGenerator.java]: That's a great suggestion, thanks. Changed in https://github.com/Unity-Technologies/android-jni-bridge/pull/41/commits/ff40b49e4ecef4d81045459090bd51f10888a735

File: apigenerator/APIGenerator.java (hunk 4 above)
Comment (juliusmik) [reply apigenerator/APIGenerator.java]: I don't know why it's done like this. I've switched non-error message to be printed to System.out in https://github.com/Unity-Technologies/android-jni-bridge/pull/41/commits/ff40b49e4ecef4d81045459090bd51f10888a735 One immediate change that I've noticed is that in the log first we see all of the system.out messages and then we get system.err messages, where previously everything was "in order". But I think it's fine to have errors listed at the end, after the normal output.

**Comment by juliusmik** [comment]
Waiting for the tests to run before merging this PR.

**Pull request merged**